daemons/controld/controld_join_client.c

```diff
@@ -267,6 +267,12 @@ do_cl_join_finalize_respond(long long action,
 
     update_dc_expected(input->msg);
 
+    /* record the node's feature set as a transient attribute */
+    cib__update_node_attr(logger_out, fsa_cib_conn,
+                          cib_sync_call | cib_can_create, XML_CIB_TAG_STATUS,
+                          fsa_our_uuid, NULL, NULL, NULL, CRM_ATTR_FEATURE_SET,
+                          CRM_FEATURE_SET, NULL, NULL);
```

Comment:
I prefer to go through pacemaker-attrd for transient attributes. The controller has a convenience function for it, update_attrd()

Reply:
Changed to `updated_attrd()` in the updated patch.